### ISSUE TITLE
Add Mercury Editor for front-end layout management

### ISSUE THREAD

**Issue by finnlewis** [localgovdrupal/localgov_microsites#459]
@andybroomfield just demoed this https://www.drupal.org/project/mercury_editor

Looks great! I wonder if we might want to include it by default in localgov_microsites?

**Comment by willguv** [comment]
Initial chat about this at https://github.com/localgovdrupal/localgov/issues/711

**Comment by willguv** [comment]
@finnlewis @msayoung will release and test this

**Comment by finnlewis** [comment]
@msayoung sorry, still not got a pull request for this!

**Comment by msayoung** [comment]
@tonypaulbarker agreed to have a look at this. 
@finnlewis to point the branch he's been working on. 

- enable Mercury editor
- set permissions
- enable on appropriate fields (layout paragraphs)

config should probably go in to a submodule.

**Comment by finnlewis** [comment]
Just for the record: 

We probably want this to be a module that works with localgov proper and localgov_microsites. 

 localgov_mercury_editor  : https://www.drupal.org/project/localgov_mercury_editor 

With a sub-module for localgov_mercury_editor_microsites 

We can then make this a dependency of localgov_microsites_group, or the localgov_microsites profile

**Comment by tonypaulbarker** [comment]
Exploring scenarios, using recipes and the config orchestration. 

A few questions arising that I will look into: 
What tests do we need to write?
How does config for various content types / fields / permissions look?
Any content types that it doesn't make sense for?
How to orchestrate for bundles provided by add on modules?
Are there any gotchas like compatibility with publishing workflows (that could be a red flag)?

**Comment by tonypaulbarker** [comment]
Made some good progress here this week - nearly done with experiments and I anticipate having some working code committed very soon. I looked into writing recipes for this and I don't think it offers us a benefit for these scenarios so I expect a good ole module it shall be. 

Let's start with the gotchas.

The editing experience is quite intrusive - you either get Mercury Editor for a content type or the out of the box form, but you can't switch between them. It doesn't work well on smaller screens like a small to medium laptop because it needs an awful lot of 'real estate' - so I think being able to toggle or do something on the user settings page to turn on / off per user is something for the roadmap and may be a pre-requisite for turning it on by default.

The live edit module doesn't have a release or adoption and seems a way off being stable. So one possible 'to do' for the roadmap is to look at the deprecated live editor that comes bundled with Mercury Editor and see if that looks feasible. Without this, the module is not useful for content types that don't use Layout Paragraphs but the high risk of security vulnerabilities and bugs leads me toward starting out with 'only on' for page builder content. It was mentioned above to start out that way also. 

Now the good - 

I think the drag and drop editor is great.
One module for microsites and localgov projects seems very doable. 
Workflows seem to work well with no additional effort from us - no issue anticipated there.

I assume that only localgov_admin and microsites controller should access Mercury Editor settings that can turn it on / off for content types (and not trusted editors and editors for example).

More to come.

**Comment by willguv** [comment]
@tonypaulbarker to provide a module and demo it

**Comment by tonypaulbarker** [comment]
@finnlewis I am still working out the orchestration but aiming at a single project localgov_mercury_editor that could be added to localgov_microsites and localgov

I think the result would be the same if we required it in localgov_microsites_group instead but I don't think it will depend on group even if it has some features to support group configuration. We shall see.

Hope that helps!

**Comment by willguv** [comment]
@tonypaulbarker is continuing with this - finish the module so it can be tested

**Comment by willguv** [comment]
See "Localgov Mercury Editor" branch for latest work

**Comment by tonypaulbarker** [comment]
@finnlewis @andybroomfield @msayoung  

Please review this MR. The readme contains known issues 

Shall we tag an alpha release and start using the issue queue? 

https://git.drupalcode.org/project/localgov_mercury_editor/-/merge_requests/1

**Comment by tonypaulbarker** [comment]
@finnlewis  @msayoung and I discussed the major blocker to including in the profile, which is that I don't think there is a way to currently turn on / off for individual users, which makes this intrusive and unusable for people using smaller screens. Let's re-test and sense check that assumption - alongside testing and reviewing the other issues noted in the readme.

**Comment by andybroomfield** [comment]
Would it make sense to be able to switch between mercury editor and the normal drupal editor? From memory the mercury editor edit urls are the UUID and it serves a redirection which we could intercept with a user preference.

**Comment by tonypaulbarker** [comment]
@andybroomfield yes I think having the ability to toggle would be better still. Ideally my toggle state would persist as I move through content.

**Comment by willguv** [comment]
@finnlewis to update progress

**Comment by willguv** [comment]
@tonypaulbarker says there's an alpha release - provide a link

**Comment by tonypaulbarker** [comment]
https://www.drupal.org/project/localgov_mercury_editor
https://www.drupal.org/project/localgov_mercury_editor/releases/1.0.0-alpha1 
https://www.drupal.org/project/issues/localgov_mercury_editor 

Some interest from Nik on slack who was asking for this. I think we can use the project issue queue moving forward.
